content/scripts/build.js

```diff
@@ -601,6 +601,9 @@ class Builder {
         this.logger.info(
           chalk.yellow(`${allTitlesJsonFilepath} existed but is out-of-date.`)
         );
+
+        // We can't keep what we got from the cached disk, so clear it.
+        this.allTitles.clear();
```

Comment:
In order to cover all cases (including future cases), it seems that this should be done immediately before walking the folders. I was thinking either immediately before or after the `const allPopularities = this._getAllPopularities();` line below. Putting it here seems to me to only cover a subset of cases (the ones that try to load from cached disk), right? Or maybe I'm missing something and this does cover all practical cases?